File: LeetCode/Easy/0704-binary-search/0704-binary-search.go
func search(nums []int, target int) int {
    l, r := 0, len(nums) - 1

    for l <= r {
        mid := (l + r) / 2
        v := nums[mid]

        if v == target {
            return mid
        } else if v < target {
            l = mid + 1
        } else {
            r = mid - 1
        }
    }

    return -1
}

File: LeetCode/Easy/0704-binary-search/0704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1

        while l <= r:
            mid = (l + r) // 2
            v = nums[mid]

            if v == target:
                return mid
            elif v < target:
                l = mid + 1
            else:
                r = mid - 1
        
        return -1